auraed/src/runtime/cgroup_table.rs

```diff
@@ -24,17 +24,16 @@ impl CgroupTable {
     /// Returns an error if a duplicate [cell_name] already exists in the cache.
     pub(crate) fn insert(
         &self,
-        cell_name: &str,
-        cgroup: &Cgroup,
+        cell_name: String,
+        cgroup: Cgroup,
     ) -> Result<()> {
         let mut cache = self
             .cache
             .lock()
             .map_err(|e| anyhow!("failed to lock cgroup_table: {e:?}"))?;
         // Check if there was already a cgroup in the table with this cell name as a key.
-        if let Some(_old_cgroup) =
-            cache.insert(cell_name.into(), cgroup.clone())
-        {
+        // TODO: This seems like a bug since it potentially replaces an existing value. Guard against this scenario.
```

Comment:
I thought the insert would fail. good catch. I guess we need `entry(...).or_insert` or something.

Reply:
Yeah, the docs say it will update.